app/controllers/media_controller.rb

```diff
@@ -76,6 +74,22 @@ def requested_from_draft_assets_host?
     request.host == AssetManager.govuk.draft_assets_host
   end
 
+  def draft_asset_manager_access?
+    token = params.fetch(:token, cookies[:auth_bypass_token])
+    payload = decode_auth_bypass_token(token)
+
+    return false if payload == false
+
+    payload["draft_asset_manager_access"].present? && payload["draft_asset_manager_access"] == true
+  end
+
+protected
```

Comment:
It's somewhat unconventional to expose controller helper methods as public methods. I assume that this has been done as this is now a monster controller with bags of logic that is hard to test?

I'd suggest that you instead keep the methods protected but test them through actions that exercise the controller. 

For examples if you see in the media_controller_spec how the tests run a request and check a response: https://github.com/alphagov/asset-manager/blob/10537aa1437a18b3416e61c86c9bf19093bd3c44/spec/controllers/media_controller_spec.rb#L526-L528